pkg/ebpf/probes.go

```diff
@@ -0,0 +1,72 @@
+// +build linux_bpf
+
+package ebpf
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/ebpf/manager"
+)
+
+const x64SyscallPrefix = "__x64_"
+
+// ChooseSyscallProbeExit chooses between a tracepoint or kretprobe based on configuration and
+// host-supported features.
+func (c *Config) ChooseSyscallProbeExit(tracepoint string, fallback string) (string, error) {
```

Comment:
Curious, why `ChooseSyscallProbeExit` and `ChooseSyscallProbe` are `Config` methods?

Reply:
They use `EnableTracepoints` on `Config` at the moment.

Reply:
The config could easily be an argument though.